presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1957,11 +1957,15 @@ private PhysicalOperation createLookupJoin(
 
             OperatorFactory operator = createLookupJoin(node, probeSource, probeSymbols, probeHashSymbol, lookupSourceFactory, context, spillEnabled);
 
+            // LookupJoinOperator returns probe data on the left and build data on the right.
+            // Need to map output symbols accordingly to the actual order of channels.
             ImmutableMap.Builder<Symbol, Integer> outputMappings = ImmutableMap.builder();
-            List<Symbol> outputSymbols = node.getOutputSymbols();
```

Comment:
I find it confusing that `node.getOutputSymbols()` is still a list. It should be changed to `Set<Symbol>` or `Map<Symbol, Integer>` instead so that it's explicit that order of symbols in `getOutputSymbol` is meaningful.

Alternatively, we could have `getOutputSymbols` to return `Set<Symbol>`, and add method:
`Map<Symbol, Integer> OperatorFactory#getLayout()`.